src/components/SafeAreaConsumer.tsx

```diff
@@ -1,29 +1,34 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import {SafeAreaInsetsContext} from 'react-native-safe-area-context';
+import {EdgeInsets, SafeAreaInsetsContext} from 'react-native-safe-area-context';
+import {DimensionValue} from 'react-native';
 import * as StyleUtils from '../styles/StyleUtils';
 
-const propTypes = {
-    /** Children to render. */
-    children: PropTypes.oneOfType([PropTypes.func, PropTypes.node]).isRequired,
+type ChildrenProps = {
+    paddingTop?: DimensionValue;
+    paddingBottom?: DimensionValue;
+    insets?: EdgeInsets;
+    safeAreaPaddingBottomStyle: {
+        paddingBottom?: DimensionValue;
+    };
+};
+
+type SafeAreaConsumerProps = {
+    children: (props: ChildrenProps) => React.ReactNode;
```

Comment:
Seems like there should be a type for "Single children with specific props" somewhere, maybe in react-native itself?